src/w_wad.c

```diff
@@ -140,6 +140,9 @@ wad_file_t *W_AddFile (const char *filename)
 	return NULL;
     }
 
+    // [crispy] save the file name
+    wad_file->basename = M_BaseName(wad_file->path);
```

Comment:
Since this should always be a pure transformation of wad_file->path, can we just call `M_BaseName()` when the basename is needed instead? Additional state that can get out of sync is a code smell to me.